### ISSUE TITLE
API ModelQueries.list does not create filter correctly when querying by parent ID

### ISSUE THREAD

**Issue by ragingsquirrel3** [aws-amplify/amplify-flutter#1419]
### Description

If you have a model with a parent (e.g. posts that belong to a blog) and you try to use model helpers to make list query like:

```dart
final req =
            ModelQueries.list(Post.classType, where: Post.BLOG.eq(blogId));
 final res = await Amplify.API.query(request: req).response;
```

Then you get a graphql error from appsync like:

```
GraphQLResponseError{
  "message": "The variables input contains a field name 'blog' that is not defined for input object type 'ModelPostFilterInput' "
}
```

That is because the model helper incorrectly translates the query predicate to a filter, and uses "blog" as the name of the field instead of targetName in the schema, in this case "blogID." Note there is no BLOGID query field in the schema that can be used here in the filter instead.

### Categories

- [ ] Analytics
- [X] API
- [ ] Auth
- [ ] DataStore
- [ ] Storage

### Steps to Reproduce

see above

### Platforms

- [X] iOS
- [X] Android

### Environment

```bash
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 2.8.1, on Mac OS X 10.15.7 19H1715 darwin-x64, locale en-US)
[!] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    ✗ cmdline-tools component is missing
      Run `path/to/sdkmanager --install "cmdline-tools;latest"`
      See https://developer.android.com/studio/command-line for more details.
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/macos#android-setup for more details.
[!] Xcode - develop for iOS and macOS (Xcode 12.4)
    ! Flutter recommends a minimum Xcode version of 13.0.0.
      Download the latest version or update via the Mac App Store.
[✓] Chrome - develop for the web
[✓] Android Studio (version 4.1)
[✓] VS Code (version 1.64.2)
[✓] Connected device (3 available)

! Doctor found issues in 2 categories
```

### Dependencies

```bash
not needed
```

### Device

not needed

### OS

not needed

### CLI Version

not needed

**Comment by filippomenchini** [comment]
Hi! Do you have an idea on when this fix should be released? Thanks!

**Comment by ragingsquirrel3** [comment]
Just released w 0.4.2
